### ISSUE TITLE
Change Request: Global ignores helper

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#19403]
### ESLint version

HEAD

### What problem do you want to solve?

Help people to better distinguish between global and local ignores, more specifically, helping ensure that they don't add additional keys into a config object that is intended as a global ignores object.

(Similar to https://github.com/eslint/eslint/issues/19154)

### What do you think is the correct solution?

Because we are already creating the `@eslint/config-helpers` package to export `defineConfig()`, we can also export a `globalIgnores()` function to help make it clear that people are intending to globally ignore some patterns. For example:

```js
import { defineConfig, globalIgnores } from "@eslint/config-helpers";
import js from "@eslint/js";

export default defineConfig([
    {
        plugins: { js },
        extends: ["js/recommended"],
        rules: {
            "no-console": "error"
        }
    },
    globalIgnores(["dist", "tmp"])
]);
```

The `globalIgnores()` would just produce:

```js
{
    name: "Global ignores",
    ignores: ["dist", "tmp"]
}
```

So it would not actually change any syntax, it just acts as a wrapper that ensures no additional keys are set on the same object.

### Participation

- [x] I am willing to submit a pull request for this change.

**Comment by mdjermanovic** [comment]
Good idea, makes sense to me 👍

**Comment by JoshuaKGoldberg** [comment]
👍 I very much like this.

Should it allow customizing the name? `globalIgnores(["dist", "tmp"], "Generated files")`?

I also wonder if this establishes a precedent that other config helpers will want to use. If so, which ones? None come to mind at the moment.

**Comment by mdjermanovic** [comment]
Fixed by https://github.com/eslint/rewrite/pull/159 and https://github.com/eslint/eslint/pull/19487.
